sqlengine/postgres_engine.go

```diff
@@ -37,6 +40,18 @@ func (d *PostgresEngine) Open(address string, port int64, dbname string, usernam
 	return nil
 }
 
+func (d *PostgresEngine) Ping() error {
+	err := d.db.Ping()
+	if err != nil {
+		d.logger.Error("sql-ping-error", err)
+		if strings.Contains(err.Error(), "password authentication failed") {
+			return errors.New("Login failed")
```

Comment:
It would be better to define a package-level variable for this error, and return that. That way a caller can test against that error instead of having to do string matching.